controllers/studentController.js

```diff
@@ -104,3 +104,36 @@ exports.pickRP = async (req, res) => {
   }
 };
 
+exports.allStudent = async (req, res) => {
+  try {
+    const studentList = await Users.find({ role: "student" });
+    const page = parseInt(req.query.page);
+    const limit = parseInt(req.query.limit);
+
+    const startIndex = (page - 1) * limit;
+    const endIndex = page * limit;
+
+    const results = {};
+
+    if (endIndex < await studentList.length) {
```

Comment:
This is nice.
but....awaiting can work inside here
perhaps it has been awaited in line 109.
Let me know why you did it like that.

Good one beejay

Reply:
That was  not suppose to be their, i skipped it